Makefile

```diff
@@ -77,6 +78,16 @@ stacker-cov: $(STAGE1_STACKER) $(STACKER_DEPS) cmd/stacker/lxc-wrapper/lxc-wrapp
 		--substitute VERSION_FULL=$(VERSION_FULL) \
 		--substitute WITH_COV=yes
 
+# On Ubuntu 24.04 the lxc package does not link against libsystemd so the pkg-config
+# below does list -lsystemd; we must add it to the list but only for stacker-dynamic
+ifeq ($(shell awk -F= '/VERSION_ID/ {print $$2}' /etc/os-release),"24.04")
+ifeq (stacker-dynamic,$(firstword $(MAKECMDGOALS)))
+LXC_WRAPPER_LIBS=-lsystemd
+else
+LXC_WRAPPER_LIBS=
```

Comment:
Hm, it probably doesn't matter as I don't think it'll fail on undefined variables, but I think you meant for LXC_WRAPPER_LIBS= to also happen if it's not 24.04?

Reply:
Actually I don't.  On 24.04, the `lxc-wrapper` binary requires a static link to systemd;

```
$ make stacker-dynamic
make -C cmd/stacker/lxc-wrapper LDFLAGS=-static LDLIBS="-llxc -lutil -lpthread -ldl -lpam -lseccomp -lselinux -lsepol -lpcre2-8 -L/usr/lib64 -lcap   -lpthread -ldl" lxc-wrapper
make[1]: Entering directory '/home/ubuntu/stacker/cmd/stacker/lxc-wrapper'
cc   -static  lxc-wrapper.c  -llxc -lutil -lpthread -ldl -lpam -lseccomp -lselinux -lsepol -lpcre2-8 -L/usr/lib64 -lcap   -lpthread -ldl -o lxc-wrapper
/usr/bin/ld: /usr/lib/gcc/x86_64-linux-gnu/13/../../../x86_64-linux-gnu/liblxc.a(conf.c.o): in function `suggest_default_idmap':
(.text+0x18eb0): warning: Using 'getgrgid_r' in statically linked applications requires at runtime the shared libraries from the glibc version used for linking
/usr/bin/ld: /usr/lib/gcc/x86_64-linux-gnu/13/../../../x86_64-linux-gnu/liblxc.a(attach.c.o): in function `lxc_attach_run_shell':
(.text+0x6486): warning: Using 'getpwuid_r' in statically linked applications requires at runtime the shared libraries from the glibc version used for linking
/usr/bin/ld: /usr/lib/gcc/x86_64-linux-gnu/13/../../../x86_64-linux-gnu/liblxc.a(cgroups_cgfsng.c.o): in function `systemd_jobremoved_callback':
(.text+0x2eeb): undefined reference to `sd_bus_message_read'
/usr/bin/ld: /usr/lib/gcc/x86_64-linux-gnu/13/../../../x86_64-linux-gnu/liblxc.a(cgroups_cgfsng.c.o): in function `cgroup_attach_move_into_leaf.isra.0':
(.text+0x6813): undefined reference to `sd_bus_open_user'
/usr/bin/ld: (.text+0x6851): undefined reference to `sd_bus_message_new_method_call'
/usr/bin/ld: (.text+0x688c): undefined reference to `sd_bus_message_append'
/usr/bin/ld: (.text+0x68b5): undefined reference to `sd_bus_call'
/usr/bin/ld: (.text+0x68d5): undefined reference to `sd_bus_message_unref'
/usr/bin/ld: (.text+0x68e6): undefined reference to `sd_bus_message_unref'
/usr/bin/ld: (.text+0x68ee): undefined reference to `sd_bus_error_free'
/usr/bin/ld: (.text+0x68ff): undefined reference to `sd_bus_unref'
/usr/bin/ld: /usr/lib/gcc/x86_64-linux-gnu/13/../../../x86_64-linux-gnu/liblxc.a(cgroups_cgfsng.c.o): in function `initialize_cgroups':
(.text+0xc7bc): undefined reference to `sd_bus_open_user'
/usr/bin/ld: (.text+0xc7fa): undefined reference to `sd_bus_call_method_async'
/usr/bin/ld: (.text+0xc852): undefined reference to `sd_bus_match_signal'
/usr/bin/ld: (.text+0xc86a): undefined reference to `sd_event_new'
/usr/bin/ld: (.text+0xc887): undefined reference to `sd_bus_attach_event'
/usr/bin/ld: (.text+0xc8a2): undefined reference to `sd_event_unref'
/usr/bin/ld: (.text+0xc90d): undefined reference to `sd_bus_unref'
/usr/bin/ld: (.text+0xe127): undefined reference to `sd_bus_message_new_method_call'
/usr/bin/ld: (.text+0xe159): undefined reference to `sd_bus_message_append'
/usr/bin/ld: (.text+0xe180): undefined reference to `sd_bus_message_open_container'
/usr/bin/ld: (.text+0xe1ea): undefined reference to `sd_bus_message_append'
/usr/bin/ld: (.text+0xe209): undefined reference to `sd_bus_message_close_container'
/usr/bin/ld: (.text+0xe22f): undefined reference to `sd_bus_message_append'
/usr/bin/ld: (.text+0xe258): undefined reference to `sd_bus_call'
/usr/bin/ld: (.text+0xe27c): undefined reference to `sd_bus_message_read'
/usr/bin/ld: (.text+0xe2a0): undefined reference to `sd_event_run'
/usr/bin/ld: (.text+0xe2d8): undefined reference to `sd_event_run'
/usr/bin/ld: (.text+0xe327): undefined reference to `sd_bus_message_unref'
/usr/bin/ld: (.text+0xe338): undefined reference to `sd_bus_message_unref'
/usr/bin/ld: (.text+0xe340): undefined reference to `sd_bus_error_free'
/usr/bin/ld: /usr/lib/gcc/x86_64-linux-gnu/13/../../../x86_64-linux-gnu/liblxc.a(cgroups_cgfsng.c.o): in function `cgroup_attach_move_into_leaf.isra.0.cold':
(.text.unlikely+0x22a): undefined reference to `sd_bus_error_free'
/usr/bin/ld: (.text.unlikely+0x292): undefined reference to `sd_bus_message_unref'
/usr/bin/ld: (.text.unlikely+0x2aa): undefined reference to `sd_bus_message_unref'
/usr/bin/ld: (.text.unlikely+0x2e1): undefined reference to `sd_bus_unref'
/usr/bin/ld: /usr/lib/gcc/x86_64-linux-gnu/13/../../../x86_64-linux-gnu/liblxc.a(cgroups_cgfsng.c.o): in function `initialize_cgroups.cold':
(.text.unlikely+0x6a1): undefined reference to `sd_bus_unref'
/usr/bin/ld: (.text.unlikely+0x707): undefined reference to `sd_bus_message_unref'
/usr/bin/ld: (.text.unlikely+0x718): undefined reference to `sd_bus_message_unref'
/usr/bin/ld: (.text.unlikely+0x720): undefined reference to `sd_bus_error_free'
collect2: error: ld returned 1 exit status
make[1]: *** [<builtin>: lxc-wrapper] Error 1
make[1]: Leaving directory '/home/ubuntu/stacker/cmd/stacker/lxc-wrapper'
make: *** [Makefile:104: cmd/stacker/lxc-wrapper/lxc-wrapper] Error 2
```


 and `pkg-config --static --libs lxc` does not emit an `-lsystemd`. 

However on 22.04; it doesn't seem to require the `-lsystemd` ;  I'm not exactly sure why; I assume it's related to how the `liblxc.a` is built in the `lxc-dev` package in 24.04 vs what's in 22.04?

The intention then is to append the `-lsystemd` when making `stacker-dynamic` but to omit the link when we create `stacker-static` which is compiled inside a stacker root we constructed, and there, we don't use/need systemd since we compile our own lxc lib.